test/acceptance/copy_test.rb

```diff
@@ -0,0 +1,505 @@
+require 'test_helper'
+require 'aws-sdk-states'
+require 'json'
+
+step_functions = Aws::States::Client.new(
+  region: 'us-east-1'
+)
+
+describe :porter do
+  describe :copy do
+    describe :to_key_with_spaces do
+      it 'handles HTTP source files with spaces' do
+        req = step_functions.start_execution({
+          state_machine_arn: CONFIG.PORTER_STATE_MACHINE_ARN,
+          input: {
+            Job: {
+              Id: 'porter-test-copy-http-spaces-to-spaces',
+              Source: {
+                Mode: 'HTTP',
+                URL: 'https://dovetail-cdn-staging.prxu.org/815/dAsUFiIz68rzpns5Cv7XM36Pt8bBzbu7gn3ui6Hg8-o/input file with spaces.mp3'
```

Comment:
I mean, technically this is an invalid URL.

Reply:
Do you think Porter should reject it?

Reply:
If we don't, worst case is it fails, best case it gets encoded to what they actually wanted.

Reply:
Nah- I think attempting to handle it is better than nothing.  Which is what you're doing.  Resolved!